### DISCUSSION TITLE
why is Canvas moved to left top - thus not centered?

### DISCUSSION THREAD

**Discussion by divinitas-art** [excaliburjs/Excalibur#2402] [General]
I am creating my own graphics using Canvas graphical component and the final image is not centered as any other graphics i use is. 
Is this some sort of a bug or does this have a reason? 

It might be a bug, since it does it in example on this page too 
https://excaliburjs.com/docs/canvas

according to the code shown, the red box should be in center of scene? and it isnt...

**Comment by eonarheim** [comment]
Hi @divinitas-art this is a great question!

The reason the red rectangle isn't centered in the example is the underlying HTML canvas under the `ex.Canvas` graphic has a default size and that canvas is centered. You can see the graphics bounds in yellow if I toggle debug mode on that snippet. (This isn't made very clear in the docs so I'll update that as well)

<img width="276" alt="image" src="https://user-images.githubusercontent.com/612071/177772492-f5f81b12-d250-40a1-8abb-fb005f21d4b0.png">

However there does seem to be a bug in the current `ex.Canvas` which forces the dimensions to be the next greatest power of 2 size. We should be able to fix `ex.Canvas` to allow arbitrary width/height before we release the next version of excalibur this week 👍 

Here is a workaround to center the ex.Canvas in that snippet:
```typescript
const canvas = new ex.Canvas({
    width: 256,
    height: 256,
    cache: true,  // If true draw once until flagged dirty again, otherwise draw to Canvas every frame
    draw: (ctx) => {
        console.log('With cache=true I draw once');
        ctx.fillStyle = 'red';
        ctx.fillRect(56/2, 56/2, 200, 200);
    }
})
```
<img width="553" alt="image" src="https://user-images.githubusercontent.com/612071/177775081-7355aa41-4876-4660-ab43-1ab9f8dcf8f1.png">

**Reply by divinitas-art** [reply]
oh... now it all makes sense! yes, my canvas really is sized 1024 instead of 800, i was able to compesate for it, but i really wasnt sure why it is. 

Great, thank you very much for your answer

**Reply by eonarheim** [reply]
I've created a bug https://github.com/excaliburjs/Excalibur/issues/2403, theoretically won't be too difficult to fix

**Reply by eonarheim** [reply]
@divinitas-art The fix to this has been released as part of v0.27.0!
